climin/stops/stops.py

```diff
@@ -193,6 +194,16 @@ def __call__(self, info):
         return info['n_iter'] > self.after and info[self.key] >= self.minimal
 
 
+class IsNaN(object):
+    """Stop criterion that returns True if any value corresponding to user-specified keys is NaN."""
+
+    def __init__(self, keys=[]):
+        self.keys = keys
+
+    def __call__(self, info):
+        return any([np.isnan(info[key]) for key in self.keys])
```

Comment:
What about `IsNonFinite` with `any(~np.isfinite(info[k] for k in self.keys)`? That might capture more problems, or capture some earlier. (Thrown in from an outsider, for some reason I'm still watching this repository on github :) )